project/ServerlessDeployPlugin.scala

```diff
@@ -18,19 +36,71 @@ object ServerlessDeployPlugin extends AutoPlugin {
   override def requires: Plugins = UniversalPlugin
 
   override lazy val projectSettings = Seq(
-    serverlessDeployCommand := "serverless deploy --verbose".split(' ').toSeq,
-    deploy := Def.inputTask {
-      import scala.sys.process._
+    serverlessPackageCommand := "serverless package --verbose".split(' ').toSeq,
+    uploadToS3Command := "aws s3 cp".split(' ').toSeq,
+    cloudformationDeployCommand := "aws cloudformation deploy --capabilities CAPABILITY_NAMED_IAM".split(' ').toSeq,
+
+    prepare := Def.inputTask {
+      val stage = Stage.parser.parsed.name.toLowerCase
+      val artifactPath = (Universal / packageBin).value.toString
 
-      val baseCommand = serverlessDeployCommand.value
-      val exitCode = Process(
-        baseCommand ++ Seq("--stage", Stage.parser.parsed.name.toLowerCase),
+      val serverlessExitCode = Process(
+        serverlessPackageCommand.value ++ Seq("--stage", stage),
         Option((ThisBuild / baseDirectory).value),
-        "ARTIFACT_PATH" -> (Universal / packageBin).value.toString,
+        "ARTIFACT_PATH" -> artifactPath,
+      ).!
+
+      if (serverlessExitCode == 0) serverlessExitCode
+      else throw new IllegalStateException("Serverless returned a non-zero exit code. Please check the logs for more" +
+        " information.")
+    }.evaluated,
+
+    deploy := Def.inputTask {
+      val artifactPath = (Universal / packageBin).value.toString
+      val cloudformationTemplatePath = (ThisBuild / baseDirectory).value /
+        ".serverless" / "cloudformation-template-update-stack.json"
+
+      val updatedCloudformationTemplate = SbtIO.read(cloudformationTemplatePath).replaceAll("CloudflareLambdaFunction",
+        "Function")
+      SbtIO.write(
+        cloudformationTemplatePath,
+        updatedCloudformationTemplate
+      )
+
+      val stackNameCursor = for {
+        doc <- parser.parse(SbtIO.read((ThisBuild / baseDirectory).value / "serverless.yml"))
+        cursor = doc.hcursor
+        provider <- cursor.downField("provider").downField("stackName").as[String]
+      } yield provider
+
+      val stackName = stackNameCursor.valueOr(_ => throw new IllegalStateException("Unable to parse stack name from " +
+        "serverless.yml. Please verify it exists."))
+
+      val artifactS3PathCursor = for {
```

Comment:
This is a great use of `circe-yaml` to pull these values out of the serverless config file and not repeat them somewhere else. Nice work!